docs/runner.md

```diff
@@ -1,6 +1,159 @@
 ## Runner
 
-The runner module aims to help users to start training with less code, while stays
-flexible and configurable.
+The runner class is designed to manage the training. It eases the training process with less code demanded from users while staying flexible and configurable. The main features are as listed:
 
-Documentation and examples are still on going.
+- Support `EpochBasedRunner` and `IterBasedRunner` for different scenarios. Implementing customized runners is also allowed to meet customized needs.
+- Support customized workflow to allow switching between different modes while training. Currently, supported modes are train and val.
+- Enable extensibility through various Hooks, including hooks defined in MMCV and customized ones.
+
+## EpochBasedRunner
+
+As its name indicates, workflow in `EpochBasedRunner` should be set based on epochs. For example, [('train', 2), ('val', 1)] means running 2 epochs for training and 1 epoch for validation, iteratively. And each epoch may contain multiple iterations. Currently, MMDetection uses `EpochBasedRunner` by default.
+
+Let's take a look at its core logic:
+
+```python
+# the condition to stop training
+while curr_epoch < max_epochs:
+    # traverse the workflow.
+    # e.g. workflow = [('train', 2), ('val', 1)]
+    for i, flow in enumerate(workflow):
+        # mode(e.g. train) determines which function to run
+        mode, epochs = flow
+        epoch_runner = getattr(self, mode)
```

Comment:
may explain what epoch_runner will be, e.g., it will be self.train() or self.val()

Reply:
The part of self.call_hook can also be explained here.